File: TripKitAndroid/src/main/java/com/skedgo/tripkit/a2brouting/A2bRoutingApi.java
package com.skedgo.tripkit.a2brouting;

import java.util.List;
import java.util.Map;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;
import skedgo.tripkit.routing.RoutingResponse;

/**
 * Calculates door-to-door trips for the specified mode(s).
 * See more at https://skedgo.github.io/tripgo-api/#tag/Routing%2Fpaths%2F~1routing.json%2Fget.
 */
public interface A2bRoutingApi {
  @GET
  Observable<RoutingResponse> execute(
      @Url String url,
      @Query("modes") List<String> modes,
      @Query("avoid") List<String> excludedTransitModes,
      @Query("avoidStops") List<String> excludeStops,
      @QueryMap Map<String, Object> options
  );
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/a2brouting/A2bRoutingApi.kt
package com.skedgo.tripkit.a2brouting

import com.skedgo.tripkit.routing.RoutingResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.QueryMap
import retrofit2.http.Url

/**
 * Calculates door-to-door trips for the specified mode(s).
 * See more at https://skedgo.github.io/tripgo-api/#tag/Routing%2Fpaths%2F~1routing.json%2Fget.
 */
interface A2bRoutingApi {
    @GET
    fun execute(
        @Url url: String,
        @Query("modes") modes: List<String>,
        @Query("avoid") excludedTransitModes: List<String>,
        @Query("avoidStops") excludeStops: List<String>,
        @QueryMap options: Map<String, @JvmSuppressWildcards Any>
    ): Observable<RoutingResponse>
}